File: hutch_id_Website_OrderFlow/app/Models/Notifikasi.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Carbon\Carbon;

class Notifikasi extends Model
{
    use HasFactory;

    protected $fillable = [
        'pesanan_id',
        'tipe',
        'judul',
        'pesan',
        'data',
        'untuk_roles',
        'created_by',
        'dibaca_at',
    ];

    protected $casts = [
        'data' => 'array',
        'untuk_roles' => 'array',
        'dibaca_at' => 'datetime',
    ];

    // Accessor untuk convert created_at ke Jakarta timezone
    public function getCreatedAtAttribute($value)
    {
        if ($value) {
            return Carbon::parse($value, 'UTC')->setTimezone('Asia/Jakarta');
        }
        return $value;
    }

    public function pesanan()
    {
        return $this->belongsTo(Pesanan::class);
    }

    public function creator()
    {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function scopeNotBaca($query)
    {
        return $query->whereNull('dibaca_at');
    }
}

File: hutch_id_Mobile/lib/models/notifikasi.dart
class Notifikasi {
  final int? id;
  final String? judul;
  final String? isi;
  final String? tipe;
  final int? pesananId;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Notifikasi({
    this.id,
    this.judul,
    this.isi,
    this.tipe,
    this.pesananId,
    this.createdAt,
    this.updatedAt,
  });

  factory Notifikasi.fromJson(Map<String, dynamic> json) {
    return Notifikasi(
      id: json['id'],
      judul: json['judul'],
      isi: json['isi'],
      tipe: json['tipe'],
      pesananId: json['pesanan_id'],
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'])
          : null,
      updatedAt: json['updated_at'] != null
          ? DateTime.parse(json['updated_at'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'judul': judul,
      'isi': isi,
      'tipe': tipe,
      'pesanan_id': pesananId,
      'created_at': createdAt?.toIso8601String(),
      'updated_at': updatedAt?.toIso8601String(),
    };
  }
}
